### ISSUE TITLE
Empty Dummy Input Causes Error With Geras Renderer (possibly others)

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#2933]
**Describe the bug**

If you append an input to a block, where the the previous input is a dummy input that does not have any fields, you will recieve an error.

I have specifically found this issue with the geras renderer. Idk how to test other renderers.

I also don't know if this is actually a bug with geras, or it's just that geras doesn't support this behavior.

**To Reproduce**

I found this while writing some input tests, but the easiest place to see it (and the place it will cause the most trouble) is in the block factory.

1. Go to the block factory, on develop.
2. Add a dummy input to the block.
3. Add another input to the block (any type).
4. Observe how the block preview disappears.
5. Observe the error in the console.

**Expected behavior**

No errors, not sure what the intended render is.

**Stack Traces**

```
TypeError: Cannot read property 'isField' of undefined
    at Blockly.geras.RenderInfo.getInRowSpacing_ (blockly_compressed.js:810)
    at Blockly.geras.RenderInfo.Blockly.blockRendering.RenderInfo.addElemSpacing_ (blockly_compressed.js:756)
    at Blockly.geras.RenderInfo.Blockly.blockRendering.RenderInfo.measure_ (blockly_compressed.js:751)
    at Blockly.geras.RenderInfo.Blockly.blockRendering.RenderInfo (blockly_compressed.js:751)
    at new Blockly.geras.RenderInfo (blockly_compressed.js:809)
    at Object.Blockly.blockRendering.render (blockly_compressed.js:833)
    at Blockly.BlockSvg.render (blockly_compressed.js:924)
    at Object.BlockFactory.updatePreview (factory.js:239)
    at BlockFactory.updateLanguage (factory.js:157)
    at Blockly.WorkspaceSvg.Blockly.Workspace.fireChangeListener (blockly_compressed.js:221)
```

**Additional context**

N/A

**Comment by rachel-fenichel** [comment]
Intended appearance (from the block factory on master) for two empty dummy inputs in a row, with automatic inputs:
![image](https://user-images.githubusercontent.com/13686399/63978907-64ec5780-ca6c-11e9-9947-4e21e8c9b7e8.png)

JSON block definition:
```
{
  "type": "block_type",
  "message0": "%1 %2",
  "args0": [
    {
      "type": "input_dummy"
    },
    {
      "type": "input_dummy"
    }
  ],
  "colour": 230,
  "tooltip": "",
  "helpUrl": ""
}
```

Definitely a bug.  Even in the weirdest case, rendering shouldn't just crash.

**Comment by moniika** [comment]
This appears to be fixed in develop. An error is not thrown, although the rendering of the block has slightly changed.
![maybe-fixed](https://user-images.githubusercontent.com/6621618/65359071-33b90000-dbb0-11e9-8367-33614120aaef.png)

**Comment by RoboErikG** [comment]
The main issue is fixed, but there's still a change in rendering behavior here, we're losing a row for that second dummy input. Here's a quick comparison of with and without text in the dummy inputs. The old rendering would make the block bigger for the second input, but the new rendering seems to be deciding it has a size of 0 and falling back to the minimum block size.

![image](https://user-images.githubusercontent.com/11066279/65363480-06288280-dbc1-11e9-9f7e-46bf7e476ee7.png)
![image](https://user-images.githubusercontent.com/11066279/65363492-0de82700-dbc1-11e9-8388-e3e36ea20999.png)

Punting for now, but worth re-investigating later.

**Comment by moniika** [comment]
Investigating further into this behavior, I have found that the following things are happening, which causes some unexpected height for some blocks with empty dummy inputs:
1. If the last row has a dummy input, it is ignored (This should be fixed in #3081)
2. rows with only empty dummy input rows have height 0
3. spacer rows between empty dummy inputs and top/bottom rows have height 0
4. height of a block with one to two empty dummy input are resized to the minimum height of a block
5. the minimum height of a block < the height of an empty block

This means:
height with one dummy input = height with two dummy input < height with no dummy input < height with 3+ dummy input

This is a weird behavior but it's low priority. It only affects consecutive dummy inputs after/before top/bottom row and can be worked around by adding an additional dummy input.

Some potential solutions could be:
1. Add height to spacers between dummy input and top/bottom rows
2. Give dummy inputs a min-height (either add them as a row element with height or check hasDummyInput when computing sizing)

**Comment by samelhusseini** [comment]
Fixed by https://github.com/google/blockly/pull/3342/files#diff-609e10117f072a6024ad41c15f4dcdc5R118
